.buildkite/scripts/test-react-17.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+set -e
+
+yarn install --frozen-lockfile
+
+# Ensure that the only version of React in the repo is v17
+# so that the test suite resolves to this version
+yarn add react@17.0.2 react-dom@17.0.2 -W
+cd site && yarn add react@17.0.2 react-dom@17.0.2
```

Comment:
I wanted to be positive that there are no versions of React 16 that the components can resolve in this repo, so I have also (for this step only) updated `site` to React 17. I have verified that there are no stragglers with `yarn why` in ci.